.github/container/nsys_jax/nsys_jax/scripts/utils.py

```diff
@@ -0,0 +1,77 @@
+import contextlib
+import importlib.resources
+import os
+import pathlib
+import shlex
+import subprocess
+import sys
+
+
+def shuffle_analysis_arg(analysis):
+    """
+    Helper for parsing --nsys-jax-analysis[-arg] (nsys-jax) and --analysis[-arg]
+    (nsys-jax-combine) command line options.
+    """
+    if analysis is None:
+        return []
+    # [Script(A), Arg(A1), Arg(A2), Script(B), Arg(B1)] becomes [[A, A1, A2], [B, B1]]
+    out, current = [], []
+    for t, x in analysis:
+        if t == "script":
+            if len(current):
+                out.append(current)
+            current = [x]
+        else:
+            assert t == "arg" and len(current)
+            current.append(x)
+    if len(current):
+        out.append(current)
+    return out
+
+
+def analysis_recipe_path(script):
+    """
+    Return a context manager that yields the path to the analysis script named by
+    `script`. This can either be the name of a bundled analysis script from the
+    the installed analyses/ directory, or a filesystem path.
+    """
+    script_file = importlib.resources.files("nsys_jax").joinpath(
+        "analyses", script + ".py"
+    )
+    if script_file.is_file():
+        return script_file
+    assert os.path.exists(
+        script
+    ), f"{script} does not exist and is not the name of a built-in analysis script"
+    return contextlib.nullcontext(pathlib.Path(script))
```

Comment:
Why make this a contextmanager? As it is, this could just be a function returning the path.

Reply:
It's because the other branch (`return script_file`) above returns a context manager from `importlib.resources`. The extra indirection would allow `importlib.resources` to extract the relevant `.py` file if the module was installed as a `.zip` file, for example.